common/djangoapps/student/tests/test_views.py

```diff
@@ -380,6 +381,26 @@ def test_unfulfilled_entitlement(self, mock_course_overview, mock_pseudo_session
         self.assertIn('<div class="course-entitlement-selection-container ">', response.content)
         self.assertIn('Related Programs:', response.content)
 
+        # If an entitlement has already been redeemed by the user for a course run, do not let the run be selectable
+        enrollment = CourseEnrollmentFactory(user=self.user, course_id=unicode(mock_course_overview.return_value.id))
+        CourseEntitlementFactory.create(
+            user=self.user, course_uuid=program['courses'][0]['uuid'], enrollment_course_run=enrollment)
```

Comment:
Nit: Spacing does not match the edX Style guide.

More importantly, take a look at the factories do you need to assign the Mode here based on the entitlement? You might need to make sure they have the same modes.

Reply:
Link for the spacing nit above.

Reply:
http://edx.readthedocs.io/projects/edx-developer-guide/en/latest/style_guides/python-guidelines.html#breaking-long-lines